plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java

```diff
@@ -86,39 +86,47 @@
   private final MessageTarget messageTarget;
   private final Composite target;
   
+  /**
+   * If true, then this component is allowed to be partially-complete.
+   */
+  private final boolean partial;
+
   private final AccountSelector accountSelector;
   private final Text projectInput;
   private final Combo stagingLocationInput;
   private final Button createButton;
   private SelectFirstMatchingPrefixListener completionListener;
   private ControlDecoration stagingLocationResults;
 
-  private FetchStagingLocationsJob fetchStagingLocationsJob;
+  @VisibleForTesting
+  FetchStagingLocationsJob fetchStagingLocationsJob;
   @VisibleForTesting
   VerifyStagingLocationJob verifyStagingLocationJob;
 
   public RunOptionsDefaultsComponent(Composite target, int columns, MessageTarget messageTarget,
       DataflowPreferences preferences) {
-    this(target, columns, messageTarget, preferences, null);
+    this(target, columns, messageTarget, preferences, null, false);
   }
 
   public RunOptionsDefaultsComponent(Composite target, int columns, MessageTarget messageTarget,
-      DataflowPreferences preferences, WizardPage page) {
-    this(target, columns, messageTarget, preferences, page,
+      DataflowPreferences preferences, WizardPage page, boolean partial) {
```

Comment:
"partial" isn't clear to me. Maybe allowIncomplete or requireComplete with sense reversed?

Reply:
ok: `allowIncomplete`